src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala

```diff
@@ -150,6 +181,33 @@ class TextSentimentSuiteV4 extends TestBase with DataFrameEquality with TextKey
       data(0).get(2).toString == "negative")
     assert(data(1).get(0).toString == "positive")
   }
+
+  lazy val df2: DataFrame = Seq(
+    ("en", "Hello world. This is some input text that I love."),
+    ("fr", "Bonjour tout le monde"),
+    ("es", "La carretera estaba atascada. Había mucho tráfico el día de ayer."),
+    ("es", "La carretera estaba atascada."),
+    ("es", "Había mucho tráfico el día de ayer."),
+    ("es", "La carretera estaba atascada. Había mucho tráfico el día de ayer."),
+    (null, "ich bin ein berliner")
+  ).toDF("lang", "text")
+
+  test("Sentiment - batch usage"){
+    lazy val detector3: TextSentimentV4  = new TextSentimentV4(options)
+      .setSubscriptionKey(textKey)
+      .setEndpoint("https://eastus.api.cognitive.microsoft.com/")
+      .setInputCol("text")
+      .setLangCol("lang")
+      .setBatchSize(2)

```

Comment:
if we set batchsize as 2 here --> then ideally each row should only contain 2 input documents/sentences. So the test here should check --> whether the output is splitted into each row containing 2 documents with results.   You can check the details in val tdf = detector3.transform(df2)
      .select("output2.result.sentiment") --> without calling.collect().   Try to .show to see the details.

Reply:
Got it! Will change that. was checking by asserting length to 4 meaning first 3 batches would have 2 and last would have 1 amounting in 4 rows

Reply:
output is correct with .show() will push changes now